src/usr/local/www/firewall_rules_edit.php

```diff
@@ -295,6 +317,10 @@ function is_aoadv_used($rule_config) {
 
 	unset($input_errors);
 
+	if (!array_key_exists($_POST['ipprotocol'], $icmplookup)) {
+		$input_errors[] = gettext("The IP protocol is not recognized."); 
+	}
+
```

Comment:
Thats due to github's web UI - as mentioned above it's had some bugs introduced that makes web editing hard. More so when the extraneous chars aren't immediately visible. Will try to find and remove them!